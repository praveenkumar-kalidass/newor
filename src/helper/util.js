import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
