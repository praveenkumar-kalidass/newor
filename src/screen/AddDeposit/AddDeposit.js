import React, { useState } from 'react';
import {
  ScrollView, VStack,
} from 'native-base';

import withBackground from 'helper/withBackground';
import { Background5 } from 'component/Background';

const AddDeposit = () => {
  useState({});

  return (
    <ScrollView>
      <VStack p={5} />
    </ScrollView>
  );
};

export default withBackground({ Component: AddDeposit, Background: Background5 });
