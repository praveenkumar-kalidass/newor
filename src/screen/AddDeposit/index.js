import AddDeposit from './AddDeposit';

export default AddDeposit;
