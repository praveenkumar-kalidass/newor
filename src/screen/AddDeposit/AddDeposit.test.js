import React from 'react';

import { render, fireEvent, act } from 'test/util';
import AddDeposit from './AddDeposit';

const mockNavigation = {
  navigate: jest.fn(),
};
jest.mock('@react-navigation/native', () => {
  const reactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...reactNavigation,
    useNavigation: () => mockNavigation,
  };
});
const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});
const mockUseAsset = {
  addDeposit: jest.fn(),
};
jest.mock('api/useAsset', () => () => mockUseAsset);
const mockUseUser = {
  asset: { id: 'test_asset_id' },
};
jest.mock('provider/User/useUser', () => () => mockUseUser);

describe('AddDeposit', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<AddDeposit />);

    expect(container).toMatchSnapshot();
  });

  it('should successfully add deposit', async () => {
    mockUseAsset.addDeposit.mockResolvedValueOnce({ data: { value: 12345.00 } });

    const { getByTestId } = render(<AddDeposit />);

    fireEvent(getByTestId('deposit-input-type'), 'onValueChange', 'FIXED_DEPOSIT');
    fireEvent.changeText(getByTestId('deposit-input-initial'), '12345.00');
    fireEvent.changeText(getByTestId('deposit-input-value'), '12345.00');
    fireEvent.changeText(getByTestId('deposit-input-depository-name'), 'Test');

    await act(async () => {
      await fireEvent.press(getByTestId('deposit-submit'));
    });

    expect(mockNavigation.navigate).toHaveBeenCalledWith('ASSET');
    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });

  it('should show error toast when add deposit failed', async () => {
    mockUseAsset.addDeposit.mockRejectedValueOnce();

    const { getByTestId } = render(<AddDeposit />);

    fireEvent(getByTestId('deposit-input-type'), 'onValueChange', 'FIXED_DEPOSIT');
    fireEvent.changeText(getByTestId('deposit-input-initial'), '12345.00');
    fireEvent(getByTestId('deposit-input-same-as-initial'), 'onChange');
    fireEvent.changeText(getByTestId('deposit-input-depository-name'), 'Test');

    await act(async () => {
      await fireEvent.press(getByTestId('deposit-submit'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });
});
