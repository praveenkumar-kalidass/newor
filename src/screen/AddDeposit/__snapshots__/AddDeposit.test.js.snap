// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDeposit should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGForeignObject
              height="100%"
              width="100%"
              x={0}
              y={984}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#ad0000",
                    "fontFamily": undefined,
                    "fontSize": 200,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 300,
                    "textDecorationLine": undefined,
                  }
                }
              >
                ₹
              </Text>
            </RNSVGForeignObject>
            <RNSVGForeignObject
              height="100%"
              width="100%"
              x={525}
              y={-100}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#00238a",
                    "fontFamily": undefined,
                    "fontSize": 400,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 600,
                    "textDecorationLine": undefined,
                  }
                }
              >
                ₹
              </Text>
            </RNSVGForeignObject>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <RCTScrollView
        contentContainerStyle={
          Array [
            Object {},
            Object {
              "dataSet": Object {},
            },
          ]
        }
        dataSet={Object {}}
        style={Object {}}
      >
        <View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              }
            }
          />
        </View>
      </RCTScrollView>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
